tests/ui/impl-trait/issues/issue-78722-2.stderr

```diff
@@ -4,30 +4,21 @@ error[E0271]: expected `{async block@$DIR/issue-78722-2.rs:13:13: 13:21}` to be
 LL |         fn concrete_use() -> F {
    |                              ^ expected `()`, found `u8`
 
-error: item constrains opaque type that is not in its signature
+error[E0308]: mismatched types
```

Comment:
this diagnostics regression is unfortunate, please add a `FIXME` or handle this in this PR